src/append/rolling_file/rolling.rs

```diff
@@ -320,12 +317,9 @@ impl State {
     }
 
     fn should_rollover_on_date(&self, date: OffsetDateTime) -> bool {
-        let next_date = self.next_date;
-        if next_date == 0 {
-            false
-        } else {
-            date.unix_timestamp() as usize >= next_date
-        }
+        self.next_date_timestamp
```

Comment:
IIRC `Option` has an `is_some_and`  method to simplify these lines.

Reply:
Handled in https://github.com/tisonkun/logforth/pull/33/commits/2894986e323d5a4f9036fc79723c916f07948b60.